Service/AlgoliaConnector.php

```diff
@@ -287,26 +304,29 @@ function ($id) {
             )
         );
 
-        $this->performBatchOperation($indexName, $requests, $storeId);
+        $this->performBatchOperation($indexOptions, $requests);
     }
 
     /**
-     * @param string $fromIndexName
-     * @param string $toIndexName
-     * @param int|null $storeId
+     * @param IndexOptions $fromIndexOptions
+     * @param IndexOptions $toIndexOptions
      * @return void
      * @throws AlgoliaException
+     * @throws NoSuchEntityException
      */
-    public function moveIndex(string $fromIndexName, string $toIndexName, ?int $storeId = null): void
+    public function moveIndex(IndexOptions $fromIndexOptions, IndexOptions $toIndexOptions): void
```

Comment:
Same as for copy operations I'm thinking we should indicate that these cannot be performed across application IDs.